### PR TITLE
 fix(core/webidl): autolink IDL partial definitions

### PR DESCRIPTION
**Pull request by marcoscaceres**
xrefs things like:

```
partial interface Element {};
partial mixin Whatever {};
partial dictionary SomeDictionary {};
```

To the right specs.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Will add tests tomorrow...

File: src/core/webidl.js [hunk 1]
```diff
@@ -131,10 +132,17 @@ function createIdlAnchor(escaped, data, parent) {
   if (isDefaultJSON) {
     return hyperHTML`<a data-lt="default toJSON operation">${escaped}</a>`;
   }
+  const unlinkedAnchor = hyperHTML`<a
+    data-link-type="idl"
+    data-idl="${data.partial ? "partial" : null}"
+    data-title="${escaped}"
```
Comment (saschanaz) [inline src/core/webidl.js]: Maybe unescaped name?

```suggestion
    data-title="${data.name}"
```

File: src/core/webidl.js [hunk 2]
```diff
@@ -131,10 +132,17 @@ function createIdlAnchor(escaped, data, parent) {
...
+    data-xref-type="${getDfnType(data.type)}">${escaped}</a>`;
+
+  // Partial interfaces must always link somewhere else.
+  if (data.partial && !dfn) {
```
Comment (saschanaz) [inline src/core/webidl.js]: Might be able to merge this with `showWarnings` below.

Comment (marcoscaceres) [reply src/core/webidl.js]: I spent like an hour trying to... and I couldn't do it 😭 There are subtle differences in how they work and in the end I couldn't figure it out. 

We can maybe try again as a follow up, but I'd like to get this out as a bunch of specs are currently broken.

File: src/core/webidl.js [hunk 3]
```diff
@@ -114,27 +114,40 @@ function createIdlAnchor(escaped, data, parent) {
...
+    return unlinkedAnchor;
   }
 
-  const unlinkedAnchor = hyperHTML`<a data-xref-type="${
-    data.type
-  }">${escaped}</a>`;
   const showWarnings = name && data.type !== "typedef";
```
Comment (saschanaz) [inline src/core/webidl.js]:
```suggestion
   const showWarnings = name && data.type !== "typedef" && !(data.partial && !dfn);
```

I think this should work.

File: tests/spec/core/webidl-spec.js [hunk 4]
```diff
@@ -1218,4 +1218,79 @@ callback CallBack = Z? (X x, optional Y y, /*trivia*/ optional Z z);
...
+    `;
+    const ops = makeStandardOps({ xref: "web-platform" }, body);
+    const doc = await makeRSDoc(ops);
+    const [
+      banana,
+      bananaPartial,
+      docOrShadowMixin,
+      requestPartialInterface,
+      eventInitDict,
+      ,
```
Comment (saschanaz) [inline tests/spec/core/webidl-spec.js]:
```suggestion
```

A mistake?

Comment (marcoscaceres) [reply tests/spec/core/webidl-spec.js]: nope, that one skips "boolean"

Comment (marcoscaceres) [reply tests/spec/core/webidl-spec.js]: Can you suggest a comment // skip "boolean" link?

Comment (saschanaz) [reply tests/spec/core/webidl-spec.js]: Ah, oops, right, sorry!

File: tests/spec/core/webidl-spec.js [hunk 5]
```diff
@@ -1248,8 +1248,8 @@ callback CallBack = Z? (X x, optional Y y, /*trivia*/ optional Z z);
       bananaPartial,
       docOrShadowMixin,
       requestPartialInterface,
-      eventInitDict,
-      , // skip testing boolean link, tested elsewhere.
+      eventInitDict, // skip testing boolean link (next line), tested elsewhere.
```
Comment (saschanaz) [inline tests/spec/core/webidl-spec.js]: 🤔, okay eslint, so be it...

**Pull request merged**